src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2ApplicationSummaryEntity.kt

```diff
@@ -0,0 +1,115 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity
+
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.Table
+
+@Repository
+interface ApplicationSummaryRepository : JpaRepository<Cas2ApplicationSummaryEntity, String> {
+//  @Query(countQuery =
+//    """
+//    SELECT COUNT(*)
+//      FROM cas_2_applications a
+//    WHERE a.created_by_user_id = :userId
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByUserId(userId: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//  """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    WHERE a.created_by_user_id = :userId
+//    AND a.submitted_at IS NOT NULL
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByUserIdAndSubmittedAtIsNotNull(userId: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//  """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    WHERE a.created_by_user_id = :userId
+//    AND a.submitted_at IS NULL
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByUserIdAndSubmittedAtIsNull(userId: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//    """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    WHERE a.referring_prison_code = :prisonCode
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByPrisonCode(prisonCode: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//  """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    WHERE a.referring_prison_code = :prisonCode
+//    AND a.submitted_At IS NOT NULL
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByPrisonCodeAndSubmittedAtIsNotNull(prisonCode: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//  """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    WHERE a.referring_prison_code = :prisonCode
+//    AND a.submitted_At IS NULL
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findByPrisonCodeAndSubmittedAtIsNull(prisonCode: String, pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+
+//  @Query(countQuery =
+//  """
+//    SELECT COUNT(*)
+//      FROM cas_2_application_summary a
+//    AND a.submitted_At IS NOT NULL
+//    """,
+//    nativeQuery = true,
+//  )
+  fun findBySubmittedAtIsNotNull(pageable: Pageable?): Page<Cas2ApplicationSummaryEntity>
+}
+
+@Entity
+@Table(name = "cas_2_application_summary")
+data class Cas2ApplicationSummaryEntity(
+  @Id
+  val id: String,
```

Comment:
Could change to UUID type.

Reply:
feels like it should be a UUID right? 
It looks like we do a CAST on sql queries elsewhere in the repo `CAST(a.id AS TEXT) as id` but I don't really know why? Maybe something to do with the `ApplicationSummary` interface, or the need to convert it between psql land and jpa land?

Reply:
I've left these as is.  Changing these three CASTed strings back to UUID types gives org.springframework.dao.InvalidDataAccessApiUsageException, the cause of which I haven't been able to resolve.  For the same of getting this PR over the line I think it's acceptable as it stands.

Reply:
I think that it would be less messy to have this as a UUID - I believe that the Kotlin side is clever enough to know that this is a UUID, even though we are doing that cast in the psql query, and I've just tried locally and it seems to work. 
(You'll see in the old code that the ApplicationSummary interface had a `getID` function used on the transformer that returned a UUID even though we were CASTing to string there too.)

Reply:
Have changed `id` here from String to UUID.